src/components/textToSpeech/js/TextToSpeech.js

```diff
@@ -121,102 +123,63 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
+                funcName: "fluid.textToSpeech.invokeSpeechSynthesisFunc",
+                args: ["getVoices"]
+            }
+        },
+        listeners: {
+            "utteranceOnStart.speak": {
                 changePath: "speaking",
-                value: true
+                value: true,
+                source: "utteranceOnStart"
             },
-            // The handleEnd method is assumed to be triggered asynchronously
-            // as it is processed/triggered by the mechanism voicing the utterance.
-            handleEnd: {
+            "utteranceOnEnd.stop": {
                 funcName: "fluid.textToSpeech.handleEnd",
                 args: ["{that}"]
             },
-            handleError: "{that}.events.onError.fire",
-            handlePause: {
+            "utteranceOnError.forward": "{that}.events.onError",
+            "utteranceOnPause.pause": {
                 changePath: "paused",
-                value: true
+                value: true,
+                source: "utteranceOnPause"
             },
-            handleResume: {
+            "utteranceOnResume.resume": {
                 changePath: "paused",
-                value: false
+                value: false,
+                source: "utteranceOnResume"
+            },
+            "onCreate.unloadCleanup": {
+                funcName: "fluid.textToSpeech.cleanupOnUnload"
+            },
+            "onDestroy.cleanup": {
+                funcName: "fluid.textToSpeech.invokeSpeechSynthesisFunc",
+                args: ["cancel"]
             }
         }
     });
 
-    // Issue commands to the speechSynthesis interface with deferral (1 ms timeout);
-    // this makes the wrapper behave better when issuing commands, especially
-    // play and pause
-    fluid.textToSpeech.deferredSpeechSynthesisControl = function (control, args) {
-        setTimeout(function () {
-            speechSynthesis[control](args);
-        }, 1);
-    };
-
-    // Throttle implementation adapted from underscore.js 1.8.3; see
-    // file header for license details
-    // Returns a version of a function that will only be called max once
-    // every "wait" MS
-    fluid.textToSpeech.throttle = function (func, wait, options) {
-        var timeout, context, args, result;
-        var previous = 0;
-        if (!options) {
-            options = {};
-        }
-
-        var later = function () {
-            previous = options.leading === false ? 0 : new Date().getTime();
-            timeout = null;
-            result = func.apply(context, args);
-            if (!timeout) {
-                context = args = null;
-            }
+    // Cancel all synthesis when a page is unloaded.
+    // This is necessary so that the speech synthesis stops when navigating to a new page and so that paused
+    // speaking doesn't prevent new self voicing after a page reload.
+    fluid.textToSpeech.cleanupOnUnload = function () {
+        window.onbeforeunload = function () {
```

Comment:
Rather than binding to this directly, create a fluid.window.onBeforeUnload global event that this is added to as a namespaced listener - this will give other integrators some control over this policy if they need to. Preferably also bind to this declaratively rather than programmatically.